docs/guides/self_hosting.md

```diff
@@ -0,0 +1,117 @@
+| context | A user-specified deployment tag | No |
+| image.gatewayRepository | The docker repository to pull the LLM Engine gateway image from | Yes |
+| image.builderRepository | The docker repository to pull the LLM Engine endpoint builder image from | Yes |
+| image.cacherRepository | The docker repository to pull the LLM Engine cacher image from | Yes |
+| image.forwarderRepository | The docker repository to pull the LLM Engine forwarder image from | Yes |
+| image.pullPolicy | The docker image pull policy | No |
+| secrets.kubernetesDatabaseSecretName | The name of the secret that contains the database credentials | Yes |
+| serviceAccount.annotations.eks.amazonaws.com/role-arn | The ARN of the IAM role that the service account will assume | Yes |
+| service.type | The service configuration for the main LLM Engine server | No |
+| service.port | The service configuration for the main LLM Engine server | No |
+| replicaCount | The amount of replica pods for each deployment | No |
+| autoscaling | The autoscaling configuration for LLM Engine server deployments | No |
+| resources.requests.cpu | The k8s resources for LLM Engine server deployments | No |
+| nodeSelector | The node selector for LLM Engine server deployments | No |
+| tolerations | The tolerations for LLM Engine server deployments | No |
+| affinity | The affinity for LLM Engine server deployments | No |
+| datadog_trace_enabled | Whether to enable datadog tracing, datadog must be installed in the cluster | No |
+| aws.configMap.name | The AWS configurations (by configMap) for LLM Engine server deployments | No |
+| aws.configMap.create | The AWS configurations (by configMap) for LLM Engine server deployments | No |
+| aws.profileName | The AWS configurations (by configMap) for LLM Engine server deployments | No |
+| serviceTemplate.securityContext.capabilities.drop | Additional flags for model endpoints | No |
+| serviceTemplate.mountInfraConfig | Additional flags for model endpoints | No |
+| config.values.infra.k8s_cluster_name | The name of the k8s cluster | Yes |
+| config.values.infra.dns_host_domain | The domain name of the k8s cluster | Yes |
+| config.values.infra.default_region | The default AWS region for various resources | Yes |
+| config.values.infra.ml_account_id | The AWS account ID for various resources | Yes |
+| config.values.infra.docker_repo_prefix | The prefix for AWS ECR repositories | Yes |
+| config.values.infra.redis_host | The hostname of the redis cluster you wish to connect | Yes |
+| config.values.infra.s3_bucket | The S3 bucket you wish to connect | Yes |
+| config.values.llm_engine.endpoint_namespace | K8s namespace the endpoints will be created in | Yes |
+| config.values.llm_engine.cache_redis_url | The full url for the redis cluster you wish to connect | Yes |
+| config.values.llm_engine.s3_file_llm_fine_tuning_job_repository | The S3 URI for the S3 bucket/key that you wish to save fine-tuned assets | Yes |
```

Comment:
So what is the way to test making an inference call after deploying this Helm chart? How does a user update their Python client to point to the self-hosted backend? Need an example script or set of shell commands that shows this working end to end after running the Helm chart.

Reply:
Let's do this as a follow-up. It depends on the user's Load Balancer and DNS configs. I can type up instructions to do this with port forwarding, but I want to make sure the "meat" gets merged in case we don't have time for that. 

For my testing, I have been leveraging k8s port forwarding.